pkg/repro/repro.go

```diff
@@ -151,11 +151,44 @@ func Run(crashLog []byte, cfg *mgrconfig.Config, reporter report.Reporter, vmPoo
 		return nil, err
 	}
 	if res != nil {
-		ctx.reproLog(3, "repro crashed as:\n%s", string(ctx.report))
-		res.Stats = ctx.stats
+		// Save current report.
 		res.Title = ctx.title
 		res.Log = ctx.log
 		res.Report = ctx.report
+		ctx.reproLog(3, "repro crashed as:\n%s", string(ctx.report))
+
+		// Try to rerun the repro if the report is corrupted.
+		rep := reporter.Parse(ctx.log)
+		if rep == nil || rep.Corrupted {
+			for attempts := 0; attempts < 3; attempts += 1 {
+				ctx.reproLog(3, "report is corrupted, running repro again\n")
+				var crashed bool
+				if res.CRepro {
+					crashed, err = ctx.testCProg(res.Prog, res.Duration, res.Opts)
+				} else {
+					crashed, err = ctx.testProg(res.Prog, res.Duration, res.Opts)
+				}
+				if err != nil {
+					return nil, err
+				}
+				rep = reporter.Parse(ctx.log)
+				if rep != nil && !rep.Corrupted && crashed {
+					// Success, update saved report.
+					res.Title = ctx.title
```

Comment:
This duplicates the code above. This can be shorter if before populating res, we do:

```
for attempts := 0; ctx.corrupted && attempts < 3; attempts++ {
  ...
}
populate res from ctx
```

Reply:
The intent was to return the original report, if we fail to extract a noncorrupted one. But I guess it doesn't matter. Done.